File: leetcode/java/src/TwoSum.java
import java.util.Arrays;

public class TwoSum {
    public int[] solveWithTwoPointers(int[] nums, int target) {
        int[] sortedNums = Arrays.copyOf(nums, nums.length);
        int left = 0, right = sortedNums.length - 1;
        int [] answ = new int[2];
        boolean foundFirst = false;

        Arrays.sort(sortedNums);
        while(left != right) {
            if(sortedNums[left] + sortedNums[right] < target){
                left++;
            }
            else if (sortedNums[left] + sortedNums[right] > target) {
                right--;
            }
            else{
                for(int i = 0; i < nums.length; i++ ) {
                    if(nums[i] == sortedNums[left] && !foundFirst) {
                        answ[0] = i;
                        foundFirst = true;
                    }
                    else if(nums[i] == sortedNums[right]) {
                        answ[1] = i;
                    }
                }
                break;

            }
        }

        return answ;
    }
}

File: leetcode/python/src/two_sum.py
#https://leetcode.com/problems/two-sum/
from typing import List

class TwoSum:
    def solve_with_two_pointers(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = len(nums) - 1
        nums = sorted(list(enumerate(nums)),key=lambda x:x[1])

        while i < j:
            if nums[i][1] + nums[j][1] < target:
                i += 1
            elif nums[i][1] + nums[j][1] > target:
                j -= 1
            else:
                break

        return [nums[i][0],nums[j][0]]